packages/ketcher-macromolecules/src/hooks/useAttachmentPoints.ts

```diff
@@ -0,0 +1,48 @@
+import { AttachmentPointName, MonomerItemType } from 'ketcher-core';
+import { useMemo } from 'react';
+
+type Props = {
+  monomer: MonomerItemType;
+  attachmentPointsToBonds: Record<AttachmentPointName, unknown | null>;
+};
+
+export type PreparedAttachmentPointData = {
+  id: string;
+  label: string;
+  connected: boolean;
+};
+
+type ReturnType = {
+  preparedAttachmentPointsData: PreparedAttachmentPointData[];
+  connectedAttachmentPoints: string[];
+};
+
+export const useAttachmentPoints = ({
+  monomer,
+  attachmentPointsToBonds,
+}: Props): ReturnType => {
+  return useMemo(() => {
+    const preparedAttachmentPointsData: PreparedAttachmentPointData[] = [];
+    const connectedAttachmentPoints: string[] = [];
+
+    Object.entries(monomer?.props?.MonomerCaps ?? {}).forEach(([id, label]) => {
+      const connected = Boolean(attachmentPointsToBonds?.[id]);
+
+      if (connected) {
+        connectedAttachmentPoints.push(id);
+      }
+
+      const labelToRender = label === 'O' ? 'OH' : label;
```

Comment:
I remember the same rule in connections dialog. Can we use it everywhere from 1 place? F.e. move to separate function

Reply:
Added a small util function